cli/module_graph2.rs

```diff
@@ -70,6 +71,16 @@ lazy_static! {
     Regex::new(r#"(?i)\stypes\s*=\s*["']([^"']*)["']"#).unwrap();
 }
 
+type EmitResult = Result<
+  (
+    HashMap<String, String>,
+    Stats,
+    Diagnostics,
+    Option<IgnoredCompilerOptions>,
+  ),
```

Comment:
Consider adding struct for this purpose (`EmitOutput`?)

Reply:
That would make it inconsistent with the other APIs.  Maybe a `ResultInfo` that has the stats, diagnostics, and maybe_ignored_options as a struct, since that is the common bit and then a `Result<(HashMap<String, String>, ResultInfo), AnyError>`.

Reply:
SGTM 👍